sdk/src/client/node_api/error.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+/// Type alias of `Result` in Node errors
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Stronghold errors.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// The requested data was not found. (404)
+    #[error("The requested data {0} was not found.")]
+    NotFound(String),
```

Comment:
Why have this variant and not also use `ResponseError` for this case?

Reply:
Because then we can conveniently match on 404, i thought it was a bit more explicit then match on `ResponseError { code: 404, .., .. }`

Reply:
I mena, `ResponseError ` is just a way to display generic other errors, we dont actually match on it anywhere

Reply:
Where do we match on NotFound?

Reply:
Nvm, found it